extensions/wikia/Email/Controller/OtherEmailConfirmationController.class.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\EmailController;
+use Email\Check;
+
+abstract class AbstractEmailConfirmationController extends EmailController {
+
+	const LAYOUT_CSS = EmailController::NO_ADDITIONAL_STYLES;
+
+	private $confirmUrl;
+
+	public function initEmail() {
+		$this->confirmUrl = $this->request->getVal( 'confirmUrl' );
+		$this->assertValidParams();
+	}
+
+	protected function assertValidParams() {
+		if ( empty( $this->confirmUrl ) ) {
+			throw new Check( "Must pass value for confirmUrl" );
+		}
+	}
+
+	/**
+	 * A redefinition of our parent's assertCanEmail which removes assertions:
+	 *
+	 * - assertUserWantsEmail : Even if a user says they don't want email, they should get this
+	 * - assertEmailIsConfirmed : Even if a user hasn't confirmed their email address, they should get this
+	 * - assertUserNotBlocked : Even if a user is blocked they should still get these emails
+	 *
+	 * @throws \Email\Fatal
+	 */
+	public function assertCanEmail() {
+		$this->assertUserHasEmail();
+	}
+
+	/**
+	 * @template otherEmailConfirmation
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'buttonLink' => $this->confirmUrl,
+			'buttonText' => $this->getButtonText(),
+			'contentFooterMessages' => $this->getContentFooterMessages()
+		] );
+	}
+
+	abstract protected function getSummary();
+
+	protected function getButtonText() {
+		return $this->getMessage( 'emailext-emailconfirmation-button-text' )->text();
+	}
+
+	protected function getContentFooterMessages() {
+		$commonFooterMessage = $this->getCommonFooterMessages();
+		$emailSpecificFooterMessages = $this->getEmailSpecificFooterMessages();
+		return array_merge( $emailSpecificFooterMessages, $commonFooterMessage );
+	}
+
+	private function getCommonFooterMessages() {
+		return [
+			$this->getMessage( 'emailext-emailconfirmation-community-team' )->text()
+		];
+	}
+
+	abstract protected function getEmailSpecificFooterMessages();
+
+	protected static function getEmailSpecificFormFields() {
+			$form = [
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$form". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)